includes/frontend/class-wcmp-checkout.php

```diff
@@ -192,55 +192,60 @@ public function get_delivery_options_shipping_methods()
+                "deliveryEveningTitle"  => self::getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_EVENING_DELIVERY_TITLE),
+                "deliveryMorningTitle"  => self::getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_MORNING_DELIVERY_TITLE),
+                "deliveryStandardTitle" => self::getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_STANDARD_TITLE),
+                "deliveryTitle"         => self::getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_DELIVERY_TITLE),
+                "headerDeliveryOptions" => self::getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_HEADER_DELIVERY_OPTIONS_TITLE),
                 "houseNumber"           => __("House number", "woocommerce-myparcel"),
-                "onlyRecipientTitle"    => $this->getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_ONLY_RECIPIENT_TITLE),
+                "onlyRecipientTitle"    => self::getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_ONLY_RECIPIENT_TITLE),
                 "openingHours"          => __("Opening hours", "woocommerce-myparcel"),
                 "pickUpFrom"            => __("Pick up from", "woocommerce-myparcel"),
-                "pickupTitle"           => $this->getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_PICKUP_TITLE),
+                "pickupTitle"           => self::getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_PICKUP_TITLE),
                 "postcode"              => __("Postcode", "woocommerce-myparcel"),
                 "retry"                 => __("Retry", "woocommerce-myparcel"),
-                "signatureTitle"        => $this->getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_SIGNATURE_TITLE),
+                "signatureTitle"        => self::getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_SIGNATURE_TITLE),
                 "wrongHouseNumberCity"  => __("Postcode/city combination unknown", "woocommerce-myparcel"),
             ],
         ];
-        $chosenShippingMethodPrice = (float) WC()->session->get('cart_totals')['shipping_total'];
 
         foreach ($carriers as $carrier) {
             foreach (self::getDeliveryOptionsConfigMap($carrier) as $key => $setting) {
-                [$settingName, $function, $basePrice] = $setting;
+                [$settingName, $function] = $setting;
+                $addBasePrice = $setting[2] ?? false;
+
                 $value = $settings->{$function}($carrier . '_' . $settingName);
 
-                if ($basePrice) {
-                    $value = $value + $chosenShippingMethodPrice;
+                if ($addBasePrice) {
+                    $value += $chosenShippingMethodPrice;
                 }
 
                 Arr::set($myParcelConfig, 'config.' . $key, $value);
             }
         }
+
         $myParcelConfig['config']['priceStandardDelivery'] = $chosenShippingMethodPrice;
 
-        return json_encode($myParcelConfig, JSON_UNESCAPED_SLASHES);
+        return json_encode($myParcelConfig, JSON_THROW_ON_ERROR | JSON_UNESCAPED_SLASHES);
```

Comment:
like

Reply:
EA Extended moment 😍

Reply:
'JSON_THROW_ON_ERROR' is volgens mij pas beschikbaar vanaf PHP 7.3. wat gebeurd er als een klant nog de minimale PHP versie gebruikt die wij hebben beschreven in de plugin (PHP 7.1)

Reply:
Oef! Wat er gebeurt? Nou, dan throwt hij ook een error wanneer json_encode wel werkt... Ik ga ff kijken of dit door mn settings komt :/

Reply:
Yes, ik stond op php 7.3